Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewDelegator.cs

```diff
@@ -10,15 +11,62 @@ public class ItemsViewDelegator<TItemsView, TViewController> : UICollectionViewD
 		where TItemsView : ItemsView
 		where TViewController : ItemsViewController<TItemsView>
 	{
+
+		IDictionary<Type, CGSize> _sizeChache = new Dictionary<Type,CGSize>();
 		public ItemsViewLayout ItemsViewLayout { get; }
 		public TViewController ViewController { get; }
+		public IItemsViewSource ItemsSource { get; }
 
 		protected float PreviousHorizontalOffset, PreviousVerticalOffset;
 
-		public ItemsViewDelegator(ItemsViewLayout itemsViewLayout, TViewController itemsViewController)
+		public ItemsViewDelegator(ItemsViewLayout itemsViewLayout, TViewController itemsViewController, IItemsViewSource itemsViewSource)
```

Comment:
I didnt find a better way to pass the `IITemsVIewSource` to the `ItemsVIewDelegator`
Maybe there are better solutions which can be applied after a review

Reply:
i Know that there is the `UICollectionViewDataSource` which can be get/set by calling `collectionView.Source`
on a `UICollectionViewInstance` but this could change the whole xf architecture.

i tried to implement this fix/feature as defensive as possible because i do not want break anything.

Reply:
if we would assign a UICollectionViewDataSource to the ItemsViewCell then i would not have to pass the `IItemsViewSource` to the ctor of this class.
i could just call `collectionView.Source` directly in the method `ItemsViewDelegator.GetSizeForItem`